### PR TITLE
add delivery report parameter to the API

### PR DESCRIPTION
**Pull request by ibilous**

### CODE REVIEW

**Comment by AnPogrebnyak** [comment]
Have you read the [commit message convention](http://knowledgebase.vodafone.ua:8090/display/DSPMW/Commit+message+convention)?

1. For what purpose we need com.icthh.xm.tmf.ms.communication.utils.ApiMapper?
Delete com.icthh.xm.tmf.ms.communication.utils.ApiMapper and move its methods to the CommunicationMessageWrapper
2. Why we didn't added the delivery report flag to the REST API?
3. Cover the com.icthh.xm.tmf.ms.communication.utils.ApiMapper.CommunicationMessageWrapper by JUnit tests

**Comment by ibilous** [comment]
@AnPogrebnyak 
coz it is open source project where nobody agreed about this commit convention,  besides  your link available only for us. For our local code i am following it. 

1) ApiMapper here like mini factory with 2 `from` methods that convert parameters into wrapper class. Cannot say that it is so much redundant here. 
2) And why we should ? We have `characterists' exactly to not create tons of api parameters. Also Ihor S. was against new param. 
3)  ok

**Comment by AnPogrebnyak** [comment]
@ibilous 
1,2 - Miss it

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/MessagingHandler.java [hunk 1]
```diff
@@ -39,15 +41,16 @@ private void sendMessage(MessageResponse messageResponse, String topic) {
...
+        List<String> phoneNumbers = new ArrayList<>(wrapper.getPhoneNumbers());
         for (String phoneNumber : phoneNumbers) {
             try {
                 RuleResponse ruleResponse = businessRuleValidator.validate(message);
                 if (!ruleResponse.isSuccess()) {
                     failMessage(message, ruleResponse.getResponseCode(), ERROR_BUSINESS_RULE_VALIDATION);
                     return;
                 }
-                String messageId = smppService.send(phoneNumber, message.getContent(), message.getSender().getId());
+                String messageId = smppService.send(phoneNumber, message.getContent(), message.getSender().getId(), wrapper.getDeliveryReport());
```
Comment (ishkurko) [inline src/main/java/com/icthh/xm/tmf/ms/communication/messaging/MessagingHandler.java]: Are you use checkstyle - line length more than 120 symbols.

**Comment by ishkurko** [comment]
@ibilous Plz, add more tests: with different correct and not correct delivery flag.

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/MessagingHandler.java (hunk 1 above)
Comment (ibilous) [reply src/main/java/com/icthh/xm/tmf/ms/communication/messaging/MessagingHandler.java]: yes, but for some reason code format has not fix the line. Will change it.

**Pull request merged**